### ISSUE TITLE
implement an abreviated version of megapipe in Predict

### ISSUE THREAD

**Issue by mahafarhat** [farhat-lab/gentb-site#126]
Luca's Mega pipe, without assembly can be an alternative paired end fastq to vcf pipeline. Note that we should exclude the assembly step from Megapipe as this compute and time consuming and its output is not used by predict.

**Comment by mahafarhat** [comment]
making a note that we can use flatAnnotatorVAR_2.0.pl or (Martin's python version) after the pilon vcf and then process the resultant var with correct_varCONSEC.py before feeding into predict

**Comment by mahafarhat** [comment]
Definitely a task for our interns

**Comment by mahafarhat** [comment]
Task for David

**Comment by ejfresch** [comment]
The source code of megapipe is located here: https://github.com/farhat-lab/megapipe

**Comment by ejfresch** [comment]
> Luca's Mega pipe, without assembly can be an alternative paired end fastq to vcf pipeline. Note that we should exclude the assembly step from Megapipe as this compute and time consuming and its output is not used by predict.

The '--skip_assembly' option allows to run megapipe and skipping the genome assembly step.

**Comment by mahafarhat** [comment]
Just tested the new version after the commit and the pipeline is broken, failing at the SAM maker step

**Comment by mahafarhat** [comment]
Correction what is broken is the old Pipeline FASTQ Paired end pipeline.
I think we should fix it neverthess to benchmark against the new pipeline before decommissioning it.

**Comment by mahafarhat** [comment]
Still broken: got this error at the Quality control step:

Traceback (most recent call last): File "/n/groups/gentb_www/bin/qc_report.py", line 28, in <module> import atexit ImportError: No module named atexit

**Comment by mahafarhat** [comment]
moving the above re the old pipeline into its separate issue

I ran several jobs and I'm encountering several errors with the new impoved megapipe pipeline here is the run down:

several jobs stopped at the kraken step with no error provided

at the samtools sort step:
" [E::hts_open_format] Failed to open file /n/groups/gentb_www/predictData/tbdata_00000250/aligned.sam samtools sort: can't open "/n/groups/gentb_www/predictData/tbdata_00000250/aligned.sam": No such file or directory"


at the duplicate removal step:
" Lost error for run_3899_samtools-check-depth-mp"

at the check depth step:
" Could not find dependant job: run_3900_samtools-duplicate-removal-mp"

at the samtools depth step:
" Could not find dependant job: run_3902_samtools-check-depth-mp"

the fast lineage caller step
"Traceback (most recent call last): File "/n/groups/gentb_www/bin/vcf_cutter.py", line 58, in <module> main() File "/n/groups/gentb_www/bin/vcf_cutter.py", line 32, in main f = open(sys.argv[1], 'r') FileNotFoundError: [Errno 2] No such file or directory: 'fast-lineage-caller-vcf.py'"

**Comment by Davey-Hughes** [comment]
Dr. Farhat,

I looked at the records for a lot of these runs and it seems that for some reasons most of them were being processed in the same directory, which would cause conflicting file issues and may be the main reason why some errors were occurring. For example, there's a particular directory on O2 where 18 fastq.gz files are (9 pair-ends). I'm not sure why this happened, but maybe Martin knows.

This issue caused the errors at the kraken step, the samtools sort step error, the 'lost error' error and dependant [sic] job errors.

That being said, I've fixed the kraken step so it properly passes along the stderr from kraken along, and should show up in the website interface now. There are 2 main ways this step can fail: 1. kraken actually fails in some way, and the stderr message will show that, or 2. the classification was below the 90% threshold for TB, in which case I throw an error and stop the pipeline. In case 2 or on success, now the kraken step should show the couple lines that kraken throws describing the classification.

I believe I've fixed the fast lineage caller step. This was just an thing we were working on when we all had to leave, but that should be fine now. I'm running a test right now, but since the queue is busy with annotation jobs, it might take a while.
